packages/core/strapi/lib/services/entity-validator/validators.js

```diff
@@ -20,69 +26,107 @@ const composeValidators = (...fns) => (attr, { isDraft }) => {
  * @param {Object} attribute model attribute
  * @param {Object} validator yup validator
  */
-const addMinLengthValidator = ({ minLength }, validator, { isDraft }) =>
+const addMinLengthValidator = (validator, { minLength }, { isDraft }) =>
   _.isInteger(minLength) && !isDraft ? validator.min(minLength) : validator;
 
 /**
  * Adds maxLength validator
  * @param {Object} attribute model attribute
  * @param {Object} validator yup validator
  */
-const addMaxLengthValidator = ({ maxLength }, validator) =>
+const addMaxLengthValidator = (validator, { maxLength }) =>
   _.isInteger(maxLength) ? validator.max(maxLength) : validator;
 
 /**
  * Adds min integer validator
  * @param {Object} attribute model attribute
  * @param {Object} validator yup validator
  */
-const addMinIntegerValidator = ({ min }, validator) =>
+const addMinIntegerValidator = (validator, { min }) =>
   _.isNumber(min) ? validator.min(_.toInteger(min)) : validator;
 
 /**
  * Adds max integer validator
  * @param {Object} attribute model attribute
  * @param {Object} validator yup validator
  */
-const addMaxIntegerValidator = ({ max }, validator) =>
+const addMaxIntegerValidator = (validator, { max }) =>
   _.isNumber(max) ? validator.max(_.toInteger(max)) : validator;
 
 /**
  * Adds min float/decimal validator
  * @param {Object} attribute model attribute
  * @param {Object} validator yup validator
  */
-const addMinFloatValidator = ({ min }, validator) =>
+const addMinFloatValidator = (validator, { min }) =>
   _.isNumber(min) ? validator.min(min) : validator;
 
 /**
  * Adds max float/decimal validator
  * @param {Object} attribute model attribute
  * @param {Object} validator yup validator
  */
-const addMaxFloatValidator = ({ max }, validator) =>
+const addMaxFloatValidator = (validator, { max }) =>
   _.isNumber(max) ? validator.max(max) : validator;
 
 /**
  * Adds regex validator
  * @param {Object} attribute model attribute
  * @param {Object} validator yup validator
  */
-const addStringRegexValidator = ({ regex }, validator) =>
+const addStringRegexValidator = (validator, { regex }) =>
   _.isUndefined(regex) ? validator : validator.matches(new RegExp(regex));
 
+const addUniqueValidator = (validator, attr, __, model, updatedAttribute, entity) => {
```

Comment:
should we simplify the signature and pass (validator, metas, options)  with metas having, attr, attrName, entity, model etc ? this would simplify all the function in the index file too I believe

Reply:
That would be a better idea indeed. `options` would be used for the `isDraft`?

Reply:
Yes